File: api/src/main/java/api/dtos/filters/OpportunityFilterDTO.java
package api.dtos.filters;

import api.dtos.PaginateDTO;
import api.enums.OpportunityType;
import api.models.Tag;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class OpportunityFilterDTO extends PaginateDTO {
    private String name;

    private String description;

    private Long salary;

    private String companyName;

    private OpportunityType type;

    private Tag tag;

}

File: web/src/app/api/dtos/opportunity-filter.dto.ts
import { OpportunityType } from "../enums/opportunity-type";
import { Tag } from "../models/tag";
import { PaginateDTO } from "./paginate.dto";

export interface OpportunityFilterDTO extends PaginateDTO {
    name: string;
    description: string;
    salary: number;
    companyName: string;
    type: OpportunityType;
    tag: Tag;
}